cosmos/models/Workflow.py

```diff
@@ -560,10 +562,21 @@ def _process_finished_tasks(jobmanager):
     for task in jobmanager.get_finished_tasks():
         if task.NOOP or task.exit_status == 0:
             task.status = TaskStatus.successful
-            yield task
+        elif task.scheduling_error:
```

Comment:
oh, now I remember. In `JobManager.get_finished_tasks`, text fields are copied wholesale from the drm output (`job_info_dict` to the Task object:
```
            for task, job_info_dict in self.get_drm(drm).filter_is_done(list(tasks)):
                self.running_tasks.remove(task)
                for k, v in job_info_dict.items():
                    setattr(task, k, v)
```

Rather than putting logic in this copy-paste loop that inspects a field and sets TaskStatus accordingly, it seemed cleaner to put the logic here in `_process_finished_tasks()`, which also inspects fields set by the job manager and inferring the proper taskStatus from them.